Applications/FileIO/GocadIO/GocadAsciiReader.cpp

```diff
@@ -27,11 +32,38 @@ namespace Gocad
 const std::string mat_id_name = "MaterialIDs";
 const std::string eof_error = "Error: Unexpected end of file.";
 
-GocadTSurfaceReader::GocadTSurfaceReader()
+GocadAsciiReader::GocadAsciiReader()
+: _export_type(GOCAD_DATA_TYPE::ALL)
+{}
+
+GocadAsciiReader::GocadAsciiReader(GOCAD_DATA_TYPE const t)
+: _export_type(t)
+{}
+
+/// A GoCAD file may contain multiple datasets with the same name. To avoid
+/// conflicts when writing meshes, a unique id is appended to the mesh name if
+/// another dataset with the same name exists.
+void checkMeshNames(std::vector<std::unique_ptr<MeshLib::Mesh>> const& meshes)
 {
+    boost::uuids::random_generator generator;
+    std::size_t const n_meshes = meshes.size();
+    for (std::size_t i=0; i<n_meshes; ++i)
+    {
+        std::string name = meshes[i]->getName();
```

Comment:
I think the copy is not necessary here. It should be possible to store a `std::string const&`.

Reply:
:heavy_check_mark: